We get the flags from $(CXXFLAGS) and $(CFLAGS), if we don't remove them we add
them two times.
Define NDEBUG, otherwise we get pure debug code. Kfind will start in one instead
of twenty seconds with this define e.g. because of enabled caching of unknown
icons.
Debug output is not completely disabled, because QT_NO_DEBUG is not set.
--- kde4libs-4.0.66+svn790271.orig/cmake/modules/FindKDE4Internal.cmake
+++ kde4libs-4.0.66+svn790271/cmake/modules/FindKDE4Internal.cmake
@@ -896,11 +896,13 @@
    set(CMAKE_CXX_FLAGS_DEBUG          "-g -O2 -fno-reorder-blocks -fno-schedule-insns -fno-inline")
    set(CMAKE_CXX_FLAGS_DEBUGFULL      "-g3 -fno-inline")
    set(CMAKE_CXX_FLAGS_PROFILE        "-g3 -fno-inline -ftest-coverage -fprofile-arcs")
+   set(CMAKE_CXX_FLAGS_DEBIAN         "-DNDEBUG")
    set(CMAKE_C_FLAGS_RELWITHDEBINFO   "-O2 -g")
    set(CMAKE_C_FLAGS_RELEASE          "-O2 -DNDEBUG -DQT_NO_DEBUG")
    set(CMAKE_C_FLAGS_DEBUG            "-g -O2 -fno-reorder-blocks -fno-schedule-insns -fno-inline")
    set(CMAKE_C_FLAGS_DEBUGFULL        "-g3 -fno-inline")
    set(CMAKE_C_FLAGS_PROFILE          "-g3 -fno-inline -ftest-coverage -fprofile-arcs")
+   set(CMAKE_C_FLAGS_DEBIAN           "-DNDEBUG")
 
    if (CMAKE_SYSTEM_NAME MATCHES Linux)
      set ( CMAKE_C_FLAGS     "${CMAKE_C_FLAGS} -Wno-long-long -std=iso9899:1990 -Wundef -Wcast-align -Werror-implicit-function-declaration -Wchar-subscripts -Wall -W -Wpointer-arith -Wwrite-strings -Wformat-security -Wmissing-format-attribute -fno-common")
